src/components/settings/security/TwoFactor/AddU2FModal/index.js

```diff
@@ -0,0 +1,43 @@
+import Presentation from './presentation';
+import FormName from './formName';
+import RegisterKeyForm from './formRegisterKey';
+import { steps as RecoveryCodeSteps, beforeClose as beforeCloseRecoveryCodeModal } from '../SaveRecoveryCodeModal';
+import { steps as ScopeModal, beforeClose as beforeCloseScopeModal } from '../../../../auth/ScopeModal/index';
+import store from '../../../../../lib/store';
+import actions from '../../../../../actions/settings';
+
+export const steps = [
+    {
+        title: 'Register new U2F Key',
+        component: ({ params, onNextStep, onPreviousStep, message }) => (
+            <Presentation params={params} onSubmit={onNextStep} onCancel={onPreviousStep} message={message}/>
+        )
+    },
+    ...ScopeModal('password'),
+    {
+        title: 'Name your U2F Key',
+        component: ({ params, onNextStep, onPreviousStep }) => (<FormName
+            params={params}
+            onSubmit={onNextStep}
+            onCancel={onPreviousStep}
+        />)
+    },
+    {
+        title: 'Register new U2F Key',
+        component: ({ params, onNextStep, onPreviousStep, forbidCancel, onReset }) => (<RegisterKeyForm
+            params={params}
+            onSubmit={onNextStep}
+            onCancel={onPreviousStep}
+            forbidCancel={forbidCancel}
+            onReset={onReset}
+        />)
+    },
+    ...RecoveryCodeSteps
+];
+
+export const beforeClose = () => {
```

Comment:
What is the purpose of this function ?

Reply:
It's called after the stepped modal is closed, either by clicking the finish button, or by cancelling or clicking out of the modal area. 

It's used to reset the store for this action.